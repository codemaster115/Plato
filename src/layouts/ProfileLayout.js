import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';

import CssBaseline from '@material-ui/core/CssBaseline';
import NavBar from '../components/NavBar';

const Div = styled.div`
    background-color: rgb(230, 235, 241);
`;

const Content = styled.main`
    position: absolute;
    width: 100%;
    padding: 24px 10%;
    @media (min-width: 1280px) {
      margin-left: 240px;
      width: calc(100% - 240px);
    }
    @media (max-width: 1010px) {            
      width: 100% !important;
      padding: 24px 102px;   
    }
`;

class ProfileLayout extends React.Component {
  render() {    
    return (
      <Div>
        <CssBaseline />        
        <NavBar path={this.props.pathname} />
        <Content>
          {this.props.children}          
        </Content>      
      </Div>
    );
  }
}

ProfileLayout.propTypes = {
  children: PropTypes.object.isRequired,  
};

const mapStateToProps = state => (
  {
    pathname: state.router.location.pathname,
  }
);

const mapDispatchToProps = {
  push,
}

export default connect(mapStateToProps, mapDispatchToProps)(ProfileLayout);