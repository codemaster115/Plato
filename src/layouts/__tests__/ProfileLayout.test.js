import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import ProfileLayout from '../ProfileLayout';

const mockStore = configureStore();
let store;

describe('ProfileLayout', () => {
  beforeEach(() => {
    store = mockStore({
      router: {
        location: {},
      },
      Login: {
        username: 'test',
      },
    });
  });

  it('Compare snapshots', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <ProfileLayout>
          <div />
        </ProfileLayout>
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Renders without crashing', () => {
    mount(
      <Provider store={store}>
        <ProfileLayout>
          <div />
        </ProfileLayout>
      </Provider>,
    );
  });
});
