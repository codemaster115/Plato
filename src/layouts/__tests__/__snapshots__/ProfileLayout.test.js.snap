// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileLayout Compare snapshots 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "Login": Object {
          "username": "test",
        },
        "router": Object {
          "location": Object {},
        },
      },
    }
  }
>
  <Connect(ProfileLayout)>
    <div />
  </Connect(ProfileLayout)>
</ContextProvider>
`;
